server/chat.ts

```diff
@@ -1847,15 +1847,22 @@ export const Chat = new class {
 
 	packageData: AnyObject = {};
 
+	getPluginName(file: string) {
+		const folder = file.replace(__dirname, '').includes('chat-commands') ? 'chat-commands' : 'chat-plugins';
+		// the .slice(1, ...) removes the / from the beginning
+		let name = file.slice(file.indexOf(folder) + folder.length).slice(1, -file.lastIndexOf('.'));
+		if (name.endsWith('/index')) name = name.slice(0, -6);
+		return name;
+	}
```

Comment:
![image](https://user-images.githubusercontent.com/551184/142925228-a93fb712-9faa-4f7e-95e5-1a12b8134603.png)

Mia, please, we've talked about committing code without testing it. :(

I thought to check this because I noticed you were slicing a sliced file with the unsliced `lastIndexOf`. But I've said this repeatedly: I really don't want to be the only barrier against stuff like this hitting the repo.

Reply:
Incidentally, this is a good use for `path.relative`:

```
zarel@Asparagine ~> node
Welcome to Node.js v17.0.1.
Type ".help" for more information.
> path.relative('/foobar', '/foobar/chat-commands/asdf/asdf.ts')
'chat-commands/asdf/asdf.ts'
```

Reply:
```suggestion
	getPluginName(file: string) {
		const nameWithExt = pathModule.relative(__dirname, file).replace(/^chat-(?:commands|plugins)./, '');
		let name = nameWithExt.slice(0, nameWithExt.lastIndexOf('.'));
		if (name.endsWith('/index')) name = name.slice(0, -6);
		return name;
	}
```